lib/trino-hdfs/src/main/java/io/trino/hdfs/TrinoFileSystemCache.java

```diff
@@ -64,6 +65,8 @@
 
     public static final TrinoFileSystemCache INSTANCE = new TrinoFileSystemCache();
 
+    private static final Cleaner CLEANER = Cleaner.create();
```

Comment:
What stops the thread created here?

Reply:
Yep, the Javadoc for `Cleaner` says

> The thread runs until all registered cleaning actions have completed and the cleaner itself is reclaimed by the garbage collector.

Reply:
Can a thing (like a filesystem) not become eligible for cleanup (GC) at all, and thus Cleaner thread never completes, keeping the classloader alive?

I would feel more comfortable with something that has explicit shutdown, since we have to manage shutdown anyway.